### ISSUE TITLE
Version 0.18: TypeError: argument of type 'WindowsPath' is not iterable

### ISSUE THREAD

**Issue by screig** [xflr6/graphviz#145]
This works

```
pip install graphviz==0.17
Successfully installed graphviz-0.17
pipdeptree --graph-output svg -l --exclude pipdeptree,graphviz,setuptools,wheel,pip  > ../_Pages/dependencies_html/dependencies.svg
```

but with version 0.18 it fails

```

pip install graphviz
Successfully installed graphviz-0.18
$ pipdeptree --graph-output svg -l --exclude pipdeptree,graphviz,setuptools,wheel,pip  > ../_Pages/dependencies_html/dependencies.svg
Traceback (most recent call last):
  File "C:\Python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\GitlabRunner\builds\AEZPJs2W\0\qag\panormus_library\panormus_directdb\venv\Scripts\pipdeptree.exe\__main__.py", line 7, in <module>
  File "c:\gitlabrunner\builds\aezpjs2w\0\qag\panormus_library\panormus_directdb\venv\lib\site-packages\pipdeptree.py", line 879, in main
    is_reverse=args.reverse)
  File "c:\gitlabrunner\builds\aezpjs2w\0\qag\panormus_library\panormus_directdb\venv\lib\site-packages\pipdeptree.py", line 641, in dump_graphviz
    return graph.pipe().decode('utf-8')
  File "c:\gitlabrunner\builds\aezpjs2w\0\qag\panormus_library\panormus_directdb\venv\lib\site-packages\graphviz\piping.py", line 116, in pipe
    return self._pipe_lines(*args, input_encoding=self._encoding, **kwargs)
  File "c:\gitlabrunner\builds\aezpjs2w\0\qag\panormus_library\panormus_directdb\venv\lib\site-packages\graphviz\backend\piping.py", line 149, in pipe_lines
    proc = execute.run_check(cmd, capture_output=True, quiet=quiet, **kwargs)
  File "c:\gitlabrunner\builds\aezpjs2w\0\qag\panormus_library\panormus_directdb\venv\lib\site-packages\graphviz\backend\execute.py", line 80, in run_check
    popen = subprocess.Popen(cmd, stdin=subprocess.PIPE, **kwargs)
  File "C:\Python36\lib\subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "C:\Python36\lib\subprocess.py", line 971, in _execute_child
    args = list2cmdline(args)
  File "C:\Python36\lib\subprocess.py", line 461, in list2cmdline
    needquote = (" " in arg) or ("\t" in arg) or not arg
TypeError: argument of type 'WindowsPath' is not iterable

```

**Comment by jdavidleo** [comment]
and one of my libraries that uses graphviz is experiencing following issue:

ImportError: cannot import name 'Digraph' from 'graphviz.dot'

**Comment by xflr6** [comment]
Thanks. 

> `TypeError: argument of type 'WindowsPath' is not iterable`

Can you check if this branch would fix this for you? https://github.com/xflr6/graphviz/compare/pathlib

> `ImportError: cannot import name 'Digraph' from 'graphviz.dot'`

The library you are using might use undocumented names: import Digraph from graphviz (only import from graphviz). I am not sure if that helps. They need to pin their version and/or update to use the documented API.

**Comment by xflr6** [comment]
See also downstream issue dask/dask#8364.

Culprit is https://bugs.python.org/issue41649 (Windows-specific, fixed in Python 3.8, not backported).

**Comment by xflr6** [comment]
Pending fix in 2a72fb8322578504c7d430edc50a46cf9aba78f6

**Comment by xflr6** [comment]
Fixed in [`0.18.1` release](https://pypi.org/project/graphviz/#files), closing.
